app/javascript/mastodon/components/modal_root.js

```diff
@@ -104,7 +105,7 @@ export default class ModalRoot extends React.PureComponent {
     return (
       <div className='modal-root' ref={this.setRef}>
         <div style={{ pointerEvents: visible ? 'auto' : 'none' }}>
-          <div role='presentation' className='modal-root__overlay' onClick={onClose} style={{ backgroundColor: backgroundColor ? `rgba(${backgroundColor.r}, ${backgroundColor.g}, ${backgroundColor.b}, 0.7)` : null }} />
+          <div role='presentation' className='modal-root__overlay' onClick={onClose} style={{ backgroundColor: backgroundColor ? `rgba(${backgroundColor.r}, ${backgroundColor.g}, ${backgroundColor.b}, 0.7)` : null, transition: backgroundColor && !reduceMotion ? 'background-color 0.5s' : null }} />
```

Comment:
I think it might be better to put this into CSS for `.modal-root__overlay`. I also don't think you need to check the reduce motion setting because a color fade is not motion.